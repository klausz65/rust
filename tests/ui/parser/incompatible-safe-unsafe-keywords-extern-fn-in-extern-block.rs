//! Check that we emit a targeted suggestion for an extern fn that uses incompatible `safe` and
//! `unsafe` keywords.
#![crate_type = "lib"]

unsafe extern {
//~^ NOTE while parsing this item list starting here
    pub safe unsafe extern fn foo() {}
    //~^ ERROR expected one of `extern` or `fn`, found keyword `unsafe`
    //~| NOTE expected one of `extern` or `fn`
    //~| `safe` and `unsafe` are incompatible, use only one of the keywords
} //~ NOTE the item list ends here
