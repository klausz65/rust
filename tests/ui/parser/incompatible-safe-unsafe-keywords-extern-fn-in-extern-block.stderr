error: expected one of `extern` or `fn`, found keyword `unsafe`
  --> $DIR/incompatible-safe-unsafe-keywords-extern-fn-in-extern-block.rs:7:14
   |
LL | unsafe extern {
   |               - while parsing this item list starting here
LL |
LL |     pub safe unsafe extern fn foo() {}
   |              ^^^^^^ expected one of `extern` or `fn`
...
LL | }
   | - the item list ends here
   |
help: `safe` and `unsafe` are incompatible, use only one of the keywords
  --> $DIR/incompatible-safe-unsafe-keywords-extern-fn-in-extern-block.rs:7:9
   |
LL |     pub safe unsafe extern fn foo() {}
   |         ^^^^^^^^^^^

error: aborting due to 1 previous error

